### PR TITLE
ROX-20221: Message converters for v2 vulnerability exception service

### PR DESCRIPTION
**Pull request by md2119**
## Description
Message converters to convert `storage.Vulnerability` to `v2.VulnerabilityException` and vice-versa.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change
Unit Test

### CODE REVIEW

File: central/convert/storagetov2/vuln_exception_service.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+import (
+	"github.com/pkg/errors"
+	v2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+// VulnerabilityException converts *storage.VulnerabilityRequest to *v2.VulnerabilityException.
+func VulnerabilityException(inp *storage.VulnerabilityRequest) *v2.VulnerabilityException {
```
Comment (dashrews78) [inline central/convert/storagetov2/vuln_exception_service.go:11]: `inp` seems like an odd name.   what is that short for?

File: central/convert/storagetov2/vuln_exception_service.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	if inp.GetUpdatedDeferralReq() != nil {
+		out.UpdatedReq = &v2.VulnerabilityException_DeferralReqUpdate{
+			DeferralReqUpdate: convertDeferralReq(inp.GetUpdatedDeferralReq()),
+		}
+	}
+
+	return out
+}
+
+func convertUsers(users []*storage.SlimUser) []*v2.SlimUser {
```
Comment (dashrews78) [inline central/convert/storagetov2/vuln_exception_service.go:49]: nit:  Would it make sense to put the `user` converters in a `common.go`?

**Comment by dashrews78** [comment]
Currently converters for things like compliance and reports are with the service.  Is the idea here that we should put the storage <-> API converters in a single place `central/convert`?

File: central/convert/storagetov2/vuln_exception_service.go (hunk 1 above)
Comment (md2119) [reply central/convert/storagetov2/vuln_exception_service.go:11]: I was going for input. I think I will change it.

File: central/convert/storagetov2/vuln_exception_service.go (hunk 2 above)
Comment (md2119) [reply central/convert/storagetov2/vuln_exception_service.go:49]: you have a point

**Comment by md2119** [comment]
> Currently converters for things like compliance and reports are with the service. Is the idea here that we should put the storage <-> API converters in a single place `central/convert`?

That is correct. This approach has a few benefits over putting the converters in the service:
- All converters residing at one location makes the converter sharable for the common proto messages.
- It makes referencing the converters easier. The generic name of the package `service` often makes it hard to get the correct auto-complete swiftly.
- The package names `storagetov2` and `v2tostorage` make the functionality abundantly obvious.

**Comment by md2119** [comment]
> LGTM. Should we document somewhere (maybe the API guide) that the converters should go here and the naming scheme where common stuff is named after the type and things named `_service` handle the conversions for the service?

That is the plan. I would like to move the existing converters as well so that there is no confusion due to multiple approaches.

**Pull request merged**